krabs/krabs/filtering/event_filter.hpp

```diff
@@ -123,9 +142,14 @@ namespace krabs {
             return;
         }
 
-        if (!predicate_(record)) {
-            return;
-        }
+		if(predicate_ != nullptr) //only if predicate exists... else trust the native filtering
+		{
+			//software filtering is on 
+			if(!predicate_(record))
+			{
+				return; 
+			}
+		}//else this filter is set using native etw filtering...
```

Comment:
nit: this comment is a duplicate of the one above. suggest merging them into a single comment:
`// if predicate_ is null, assume that native filtering has been applied`

Reply:
Done